src/modules/siteRDF/process.cpp

```diff
@@ -98,6 +98,35 @@ Module::ExecutionResult SiteRDFModule::process(ModuleContext &moduleContext)
             }
         }
 
+    auto &dataRunningCN = processingData.realise<SampledData1D>("RunningCNTest", name(), GenericItem::InRestartFileFlag);
+    dataRunningCN.initialise(distanceRange_.x);
+    std::vector<double> runningCN;
+
+    // Zip over all distanceRange and calculate running CN
+    double countCN{};
+    for (const auto &&[x, currentCN] : zip(histAB.binCentres(), histAB.data().values()))
+    {
+        runningCN[x] += currentCN;
+    }
+
+    // Add data into SampledData1D
+    dataRunningCN += runningCN;
+
+    // Accumulate instantaneous binValues
+    auto instBinValues = histAB.data();
+
+    // Normalise Data
+    DataNormaliser1D normaliserInstBinValues(instBinValues);
+
+    // Normalise by A site population
+    normaliserInstBinValues.normaliseDivide(double(a.sites().size()));
+
```

Comment:
Here we go. So we have the normalised bin data in `instBinValues`, and we can convert that into the running coordination numbers in-place by doing this nice `std::transform`. Essentially we just step over each value and calculate and return a replacement value which is the running coordination number (which we keep track of the in the `sum` variable):
```suggestion
    auto sum = 0.0;
    std::transform(instBinValues.values().begin(), instBinValues.values().end(), instBinValues.values().begin(), [&](const auto &currentBin) {sum += currentBin; return sum; } );
```